src/ThemeProvider.tsx

```diff
@@ -17,6 +17,7 @@ export type ThemeProviderProps = {
   colorMode?: ColorModeWithAuto
   dayScheme?: string
   nightScheme?: string
+  preventServerClientMismatch?: boolean
```

Comment:
Start here: 1) Add opt-in feature to prevent output mismatch between server and client. If it's a client only application, they don't need to bother.